crates/rpc/rpc/src/eth/revm_utils.rs

```diff
@@ -116,33 +116,45 @@ impl FillableTransaction for TransactionSigned {
 /// Returns the addresses of the precompiles corresponding to the SpecId.
 #[inline]
 pub(crate) fn get_precompiles(spec_id: SpecId) -> impl IntoIterator<Item = Address> {
-    let spec = PrecompilesSpecId::from_spec_id(spec_id);
-    Precompiles::new(spec).addresses().into_iter().copied().map(Address::from)
+    let spec = PrecompileSpecId::from_spec_id(spec_id);
+    Precompiles::new(spec).addresses().copied().map(Address::from)
 }
 
-/// Executes the [Env] against the given [Database] without committing state changes.
-pub(crate) fn transact<DB>(db: DB, env: Env) -> EthResult<(ResultAndState, Env)>
+/// Executes the [EnvWithHandlerCfg] against the given [Database] without committing state changes.
+pub(crate) fn transact<DB>(
+    db: DB,
+    env: EnvWithHandlerCfg,
+) -> EthResult<(ResultAndState, EnvWithHandlerCfg)>
 where
     DB: Database,
     <DB as Database>::Error: Into<EthApiError>,
 {
-    let mut evm = revm::EVM::with_env(env);
-    evm.database(db);
+    let mut evm = revm::Evm::builder().with_db(db).with_env_with_handler_cfg(env).build();
     let res = evm.transact()?;
-    Ok((res, evm.env))
+    let (_, env) = evm.into_db_and_env_with_handler_cfg();
+    Ok((res, env))
 }
 
-/// Executes the [Env] against the given [Database] without committing state changes.
-pub(crate) fn inspect<DB, I>(db: DB, env: Env, inspector: I) -> EthResult<(ResultAndState, Env)>
+/// Executes the [EnvWithHandlerCfg] against the given [Database] without committing state changes.
+pub(crate) fn inspect<DB, I>(
+    db: DB,
+    env: EnvWithHandlerCfg,
+    inspector: I,
+) -> EthResult<(ResultAndState, EnvWithHandlerCfg)>
 where
     DB: Database,
     <DB as Database>::Error: Into<EthApiError>,
-    I: Inspector<DB>,
+    I: GetInspector<DB>,
 {
-    let mut evm = revm::EVM::with_env(env);
-    evm.database(db);
-    let res = evm.inspect(inspector)?;
-    Ok((res, evm.env))
+    let mut evm = revm::Evm::builder()
+        .with_db(db)
+        .with_external_context(inspector)
+        .with_env_with_handler_cfg(env)
+        .append_handler_register(inspector_handle_register)
+        .build();
```

Comment:
there's also `append_handle_register` somewhere else, not clear if this entirely different or if naming is not consistent

Reply:
misspelled